shell/platform/linux/fl_external_texture_gl.cc

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_external_texture_gl.h"
+
+#include <EGL/egl.h>
+#include <GL/gl.h>
+#include <gmodule.h>
+
+struct {
+  bool valid;
+  void (*genTextures)(GLsizei n, GLuint* textures);
+  void (*bindTexture)(GLenum target, GLuint texture);
+  void (*texParameteri)(GLenum target, GLenum pname, GLenum param);
+  void (*texImage2D)(GLenum target,
+                     GLint level,
+                     GLint internalformat,
+                     GLsizei width,
+                     GLsizei height,
+                     GLint border,
+                     GLenum format,
+                     GLenum type,
+                     const void* data);
+  void (*deleteTextures)(GLsizei n, const GLuint* textures);
+} gl;
+
+struct _FlExternalTextureGl {
+  GLuint gl_texture_id;
+  FlTextureCallback callback;
+  void* user_data;
+};
+
+G_DEFINE_TYPE(FlExternalTextureGl, fl_external_texture_gl, G_TYPE_OBJECT)
+
+static void fl_external_texture_gl_dispose(GObject* object) {
+  FlExternalTextureGl* self = FL_EXTERNAL_TEXTURE_GL(object);
+  if (gl.valid) {
+    gl.deleteTextures(1, &self->gl_texture_id);
+  }
+
+  G_OBJECT_CLASS(fl_external_texture_gl_parent_class)->dispose(object);
+}
+
+static void fl_external_texture_gl_class_init(FlExternalTextureGlClass* klass) {
+  G_OBJECT_CLASS(klass)->dispose = fl_external_texture_gl_dispose;
+}
+
+static void fl_external_texture_gl_init(FlExternalTextureGl* self) {}
+
+int64_t fl_external_texture_gl_texture_id(FlExternalTextureGl* self) {
+  return reinterpret_cast<int64_t>(self);
+}
+
+bool fl_external_texture_gl_populate_texture(
+    FlExternalTextureGl* self,
+    size_t width,
+    size_t height,
+    FlutterOpenGLTexture* opengl_texture) {
```

Comment:
Add guards: `g_return_val_if_fail(GL_IS_EXTERNAL_TEXTURE_GL(self), FALSE)`

(and other cases below).